packages/graph-editor/src/components/controls/array.tsx

```diff
@@ -0,0 +1,178 @@
+            },
+        });
+    }
+
+    if (!port.isConnected && portName === 'Input' && itemsType === ANY && !(port as Input).variadic) {
+        return (
+            <Stack direction="column" gap={3}>
+                <Select value={selectItemsType} onValueChange={setSelectItemsType}>
+                    <Select.Trigger label="Type" value={selectItemsType} />
+                    {/* @ts-expect-error */}
+                    <Select.Content css={{ maxHeight: '200px' }} position="popper">
+                        {AllSchemas.map((x, i) => (
+                            <Select.Item value={x.$id!} key={i}>
+                                {x.title || x.$id}
+                            </Select.Item>
+                        ))}
+                    </Select.Content>
+                </Select>
+                <Stack direction="row" justify="end">
+                    <Button variant="primary" onClick={setType}>
+                        Set type
+                    </Button>
+                </Stack>
+            </Stack>
+        );
+    }
+
+    if (!inputItemTypes.includes(itemsType) || port.isConnected || portName === 'Output' || (portName === 'Input' && (port as Input).variadic)) {
+        return <JSONTree data={{ items: toJS(value) }} hideRoot shouldExpandNodeInitially={() => true} />;
+    }
+
+    const getItems = () => {
+        return value.map((val: any, index: number) => {
+            switch (itemsType) {
+                case STRING:
+                case NUMBER:
+                    return (
+                        <TextInput
+                            autoFocus={autofocusIndex === index}
+                            width={'100%'}
+                            value={val}
+                            onChange={(e) => { onChange(e.target.value, index) }}
+                            trailingAction={<IconButton title="Remove item" size="small" variant="invisible" icon={<Minus />} onClick={() => removeItem(index)} />}
+                        />
+                    );
+                case COLOR:
+                    return (
+                        <ColorPickerPopover
+                            defaultOpen={autofocusIndex === index}
+                            value={val}
+                            onChange={(e) => { onColorChange(e, index) }}
+                            showRemoveButton
+                            onRemove={() => removeItem(index)}
+                        />
+                    );
+                default:
+                    return <JSONTree data={{ items: toJS(value) }} hideRoot shouldExpandNodeInitially={() => true} />;
+            }
+        });
+    }
+
+    const flexDirection = itemsType === COLOR ? 'row' : 'column';
+
+    return (
+        <Stack direction={flexDirection} gap={3} align='center' wrap css={{ background: '$bgCanvas', padding: '$3' }}>
+            {getItems()}
+            <IconButton title="Add item" icon={<Plus />} size="small" onClick={addItem} />
```

Comment:
Can we rather have it that a user can configure the value prior to adding it into the array? Otherwise the user has to have an incorrect value and then modify it afterwards.